spec/services/identity/create_spec.rb

```diff
@@ -47,7 +47,7 @@
       context "when the existing profiles belong to another identity" do
         let(:ect)                { create(:mentor, :eligible_for_funding, user:) }
         let(:exisiting_identity) { ect.participant_identities.first }
-        let!(:mentor_profile)    { create(:mentor, :eligible_for_funding, trn: ect.teacher_profile.trn, user:) }
+        let!(:mentor_profile)    { create(:mentor, :eligible_for_funding, trn: ect.teacher_profile.trn).tap { |pp| pp.update!(teacher_profile: ect.teacher_profile) } }
```

Comment:
Is there a reason this can't be 

```ruby
create(:mentor,
       :eligible_for_funding,
       trn: ect.teacher_profile.trn,
       teacher_profile: ect.teacher_profile)
```
I'm guessing it's an oddity of our factory setup?

Reply:
@peteryates yes the factory does not allow for it, hence the odd setup. Which I think is ok, since this should not happen and denotes we're dealing with legacy data.